File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/datasync/user/RemoveUser.java
package com.pubnub.api.java.endpoints.datasync.user;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.java.models.consumer.datasync.user.PNDataSyncRemoveUserResult;
import org.jetbrains.annotations.Nullable;

/**
 * @see com.pubnub.api.java.datasync.DataSync#removeUser(String)
 */
public interface RemoveUser extends Endpoint<PNDataSyncRemoveUserResult> {
    /**
     * Optional eTag for a conditional delete ({@code If-Match} header).
     */
    RemoveUser ifMatch(@Nullable String ifMatch);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/datasync/user/RemoveUser.kt
package com.pubnub.api.endpoints.datasync.user

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.datasync.user.PNDataSyncRemoveUserResult

/**
 * @see [com.pubnub.api.datasync.DataSync.removeUser]
 */
actual interface RemoveUser : Endpoint<PNDataSyncRemoveUserResult>
